// srv-cli/build/lib/logger.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

exports.default = function (opt) {
  var logs = opt.logs || './logs';

  if (!_fs2.default.existsSync(logs)) {
    _fs2.default.mkdirSync(logs);
  }

  var logger = new _winston2.default.Logger({
    transports: [new _winston2.default.transports.File({
      level: 'info',
      filename: './' + logs + '/all-logs.log',
      handleExceptions: true,
      json: true,
      maxsize: 5242880, // 5MB
      maxFiles: 5,
      colorize: false
    }), new _winston2.default.transports.Console({
      level: 'debug',
      handleExceptions: true,
      json: false,
      colorize: true
    })],
    exitOnError: false
  });

  var stream = {
    write: function write(message) {
      logger.info(message);
    }
  };

  return {
    logger: logger,
    stream: stream
  };
};

var _fs = require('fs');

var _fs2 = _interopRequireDefault(_fs);

var _winston = require('winston');

var _winston2 = _interopRequireDefault(_winston);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
